src/refda/adm/ietf_dtnma_agent.c

```diff
@@ -1003,6 +1003,68 @@ static void refda_adm_ietf_dtnma_agent_edd_odm_list(refda_edd_prod_ctx_t *ctx)
      * |START CUSTOM FUNCTION refda_adm_ietf_dtnma_agent_edd_odm_list BODY
      * +-------------------------------------------------------------------------+
      */
+    refda_agent_t *agent = ctx->prodctx->parent->agent;
+    REFDA_AGENT_LOCK(agent, );
+
+    cace_ari_tbl_t table;
+    cace_ari_tbl_init(&table, 5, 0);
+    const cace_ari_type_t obj_type = CACE_ARI_TYPE_TYPEDEF;
+
+    cace_amm_obj_ns_list_it_t ns_it;
+    for (cace_amm_obj_ns_list_it(ns_it, agent->objs.ns_list); !cace_amm_obj_ns_list_end_p(ns_it);
+         cace_amm_obj_ns_list_next(ns_it))
+    {
+
+        cace_amm_obj_ns_ptr_t *const *ns_ptr = cace_amm_obj_ns_list_cref(ns_it);
+        if (!ns_ptr)
+        {
+            continue;
+        }
+        const cace_amm_obj_ns_t *ns = cace_amm_obj_ns_ptr_ref(*ns_ptr);
+        if ((ns->model_id.intenum >= 0))
+        {
+            // ignore ADMs
+            continue;
+        }
+
+        if (!ns->org_id.has_intenum || !ns->model_id.has_intenum || !ns->model_rev.valid)
+        {
+            continue;
+        }
+
+        if (ns->obsolete)
+        {
+            CACE_LOG_DEBUG("ODM %s %d / %s %d marked as obsolete, skipping", ns->org_id.name, ns->org_id.intenum,
+                           ns->model_id.name, ns->model_id.intenum);
+            continue;
+        }
+
+        cace_ari_array_t row;
+        cace_ari_array_init(row);
+        cace_ari_array_resize(row, 5);
+
+        cace_ari_set_tstr(cace_ari_array_get(row, 0), m_string_get_cstr(ns->org_id.name), true);
+        cace_ari_set_int(cace_ari_array_get(row, 1), ns->org_id.intenum);
+        cace_ari_set_tstr(cace_ari_array_get(row, 2), m_string_get_cstr(ns->model_id.name), true);
+        cace_ari_set_int(cace_ari_array_get(row, 3), ns->model_id.intenum);
+        {
+            m_string_t buf;
+            m_string_init(buf);
+            cace_date_encode(buf, &(ns->model_rev.parts), true);
```

Comment:
Should ODMs be allowed to have well-defined revision dates? It may be better to disallow this and remove from the listing? Could treat the removal as separate PR to discuss.

Reply:
Presently, `ensure-odm` will set the revision date to be the date the ODM is added. And the above code will allow that revision to be displayed. 

This seemed to make the most sense as it allows a revision date to be tracked but ties it directly to actual ODM creation.

Open to your thoughts/suggestions on that implementation.